lib/services/services_linux.c

```diff
@@ -1168,19 +1216,31 @@ resources_os_list_ocf_agents(const char *provider)
 gboolean
 services__ocf_agent_exists(const char *provider, const char *agent)
 {
-    char *buf = NULL;
     gboolean rc = FALSE;
     struct stat st;
+    char *dirs = strdup(OCF_RA_PATH);
+    char *dir = NULL;
+    char *buf = NULL;
 
     if (provider == NULL || agent == NULL) {
         return rc;
```

Comment:
We need to either free dirs before returning here, or do the strdup() after this check